File: packages/cpp/cor_binding_test_cpp/cpp/cor_binding_test_cpp/cor_binding_test_functions.cpp
#include "cor_binding_test_functions.h"

namespace cor
{
    namespace binding_test
    {
        int CorBindingTestGlobal::add_int(int a, int b)
        {
            return a + b;
        }

        double CorBindingTestGlobal::add_double(double a, double b)
        {
            return a + b;
        }

        std::string CorBindingTestGlobal::add_string(std::string a, std::string b)
        {
            return a + b;
        }

        int CorBindingTestGlobal::overloaded(int a)
        {
            return a + 1;
        }

        RString CorBindingTestGlobal::overloaded(RString a)
        {
            return a + "1";
        }

    }

}

File: packages/cs/cor_binding_test_cs/cs/CorBindingTestFunctions.cs
using CorCppDll;
using NUnit.Framework;

namespace CorBindingTest
{
    [TestFixture]
    class CorBindingTestFunctions
    {

        [Test]
        public void AddTest()
        {
            Assert.AreEqual(CorBindingTestGlobal.AddInt(1, 3), 4);
            Assert.AreEqual(CorBindingTestGlobal.AddDouble(1.0, 3.0), 4.0);
            Assert.AreEqual(CorBindingTestGlobal.AddString("aa", "bc"), "aabc");
            Assert.AreEqual(CorBindingTestGlobal.Overloaded(2), 3);
            Assert.AreEqual(CorBindingTestGlobal.Overloaded("test"), "test1");
        }

    }
}

